src/Workspaces/Core/Portable/Workspace/Solution/SolutionChanges.cs

```diff
@@ -80,4 +82,27 @@ public IEnumerable<AnalyzerReference> GetRemovedAnalyzerReferences()
             }
         }
     }
+
+    internal IEnumerable<DocumentId> GetChangedFrozenSourceGeneratedDocuments()
+    {
+        Contract.ThrowIfNull(_newSolution.CompilationState.FrozenSourceGeneratedDocumentStates);
```

Comment:
how does the caller know this will be not null?

Reply:
They just check the property: https://github.com/dotnet/roslyn/pull/77587/files#diff-c1082184909800b4a6c84cae62ea4ae37758d99f16723417e0f25d1262b527aaR188

Can make this return an empty enumerable instead, if you prefer. I was just making a NRT warning go away :)